tools/publisher/src/subcommand/fix_manifests/validate.rs

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::fs::Fs;
+use crate::package::discover_and_validate_package_batches;
+use anyhow::{anyhow, bail, Result};
+use semver::Version;
+use std::collections::BTreeMap;
+use tracing::info;
+
+/// Validations that run before the manifests are fixed.
+///
+/// For now, this validates:
+/// - `aws-config` version number matches all `aws-sdk-` prefixed versions
+/// - `aws-smithy-` prefixed versions match `aws-` (NOT `aws-sdk-`) prefixed versions
+pub(super) fn pre_validate_manifests(versions: &BTreeMap<String, Version>) -> Result<()> {
```

Comment:
while we're at it, should we squash the next COE and check that the git repository is at the matching tag?

Reply:
Added!
```
Error: Current tag `v0.5.0-3-gf4165a00742` in the local `aws-sdk-rust` repository didn't match expected release tag `v0.5.1`
```